Nos/Views/CreateUsernameSheet.swift

```diff
@@ -0,0 +1,407 @@
+                            }
+                        }
+                        .onChange(of: usernameObserver.debouncedText) { _, newValue in
+                            Task {
+                                await verify(newValue)
+                            }
+                        }
+                        .onSubmit {
+                            Task {
+                                await verify(usernameObserver.text)
+                            }
+                        }
+                        PlainText(".nos.social")
+                            .font(.clarityTitle3)
+                            .foregroundStyle(Color.secondaryTxt)
+                    }
+                    if validationFailed {
+                        usernameAlreadyClaimedText()
+                    } else {
+                        usernameAlreadyClaimedText()
+                            .hidden()
+                    }
+
+                    Spacer(minLength: 0)
+                }
+                .padding(.horizontal, 40)
+
+                NavigationLink {
+                    ExcellentChoicePage(username: usernameObserver.text, isPresented: $isPresented)
+                } label: {
+                    if isVerifying {
+                        ZStack {
+                            ProgressView()
+                                .frame(height: .zero)
+                                .tint(Color.white)
+                            PlainText(.localizable.next)
+                                .hidden()
+                        }
+                    } else {
+                        PlainText(.localizable.next)
+                    }
+                }
+                .padding(.horizontal, 40)
+                .buttonStyle(BigActionButtonStyle())
+                .disabled(verified != true || isVerifying || invalidInput)
+            }
+        }
+        .sheetPage()
+    }
+
+    private func usernameAlreadyClaimedText() -> some View {
+        PlainText(.localizable.usernameAlreadyClaimed)
+            .font(.clarity(.medium, textStyle: .subheadline))
+            .foregroundStyle(Color.red)
+            .lineSpacing(3)
+    }
+
+    private var invalidInput: Bool {
+        usernameObserver.text.count < 3
+    }
+
+    private var validationFailed: Bool {
+        verified == false
+    }
+
+    private func verify(_ username: String) async {
+        verified = nil
+
+        guard !username.isEmpty else {
+            return
+        }
+
+        isVerifying = true
+
+        defer {
+            isVerifying = false
+        }
+
+        do {
+            verified = try await namesAPI.verify(username: username)
+        } catch {
+            Log.error(error.localizedDescription)
+        }
+    }
+}
+
+fileprivate struct ExcellentChoicePage: View {
+
+    var username: String
+    @Binding var isPresented: Bool
+    @State private var isClaiming = false
+    @State private var claimError: ClaimError?
```

Comment:
Are `isClaming` and `claimError` mutually exclusive? If so, I'd prefer a single state var enum that tells us the state -- something like the `SearchState` we use in `SearchController`.

Reply:
@martindsq I think this is my only remaining question. Do you have thoughts on this?

Reply:
Yes, I'm going to refactor so that it looks like SearchState.

Reply:
@joshuatbrown I updated this file, let me know what you think about it